### PR TITLE
Improve ref manual documentation for Secrets Manager Config and Param Store Config modules -

### PR DESCRIPTION
**Pull request by neiljbrown**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

Having recently read and used the sections of the reference manual for the Secrets Manager Config and Param Store Config modules I have made a few minor improvements to them. In summary - 

1) Various minor updates in section 2 (‘Basic Setup’) and section 3 (‘Cloud Environment’) to improve readability, correct typos and broken link.

2) Updated the relevant subsections of section ‘3. Cloud Environment’ for each of the Config modules to include an additional explanation of Spring Boot Config Import support, including that this approach is now preferred over the use of Spring Cloud bootstrap PropertySource.

## :bulb: Motivation and Context

Improve project docs.

## :green_heart: How did you test it?

Proof read rendered version of updated AsciiDoc in IDE - confirmed no typos; checked new and fixed links are valid.

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] I updated reference documentation to reflect the change
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Comment by neiljbrown** [comment]
A few of the core module's unit tests in AmazonEc2InstanceDataPropertySourceTest are failing. This is unrelated to this pull request which is limited to doc updates. Thanks.

File: docs/src/main/asciidoc/spring-cloud-aws.adoc [hunk 1]
```diff
@@ -121,21 +125,21 @@ logging.level.com.amazonaws.internal.InstanceMetadataServiceResourceFetcher=erro
 ====
 
 ==== SDK credentials configuration
-In order to make calls to the Amazon Web Service the credentials must be configured for the the Amazon SDK. Spring Cloud AWS
-provides support to configure an application context specific credentials that are used for _each_ service call for requests done
-by Spring Cloud AWS components, with the exception of the Parameter Store and Secrets Manager Configuration.
+In order to make calls to Amazon web services, credentials must be configured for the Amazon SDK. Spring Cloud AWS
```
Comment (maciejwalkowiak) [inline docs/src/main/asciidoc/spring-cloud-aws.adoc:128]: It should stay "the Amazon Web Services" as this is what AWS stands for..?

**Review by maciejwalkowiak — CHANGES_REQUESTED** [review]
Thanks a lot! One thing that I believe should be changed - otherwise it's definitely a big improvement!

File: docs/src/main/asciidoc/spring-cloud-aws.adoc (hunk 1 above)
Comment (neiljbrown) [reply docs/src/main/asciidoc/spring-cloud-aws.adoc]: Agreed. Addressed. Thanks Maciej.

**Review by maciejwalkowiak — APPROVED** [review]

**Pull request merged**